src/NSwag.CodeGeneration.CSharp/Models/CSharpOperationModel.cs

```diff
@@ -49,8 +49,9 @@ public class CSharpOperationModel : OperationModelBase<CSharpParameterModel, CSh
             _resolver = resolver;
 
             var parameters = _operation.ActualParameters.ToList();
-            if (settings.GenerateOptionalParameters)
-                parameters = parameters.OrderBy(p => !p.IsRequired).ToList();
+            if ( settings.GenerateOptionalParameters ) {
+                parameters = parameters.OrderBy( p => !p.IsRequired ).ThenBy( p => p.Default == null ).ToList();
```

Comment:
Is this really necessary? This is potentially a major breaking change for some users...